### DISCUSSION TITLE
How do I enable `hls-rename-plugin`?

### DISCUSSION THREAD

**Discussion by m4dc4p** [haskell/haskell-language-server#2239] [Q&A] [answered]
I'm using HLS 1.4, and I would love to have project wide renaming, but according to https://github.com/haskell/haskell-language-server/pull/2108 the plugin is disabled by default.

Is the plugin in 1.4? If so, how do I enable it?

Thanks for any help!

**Comment by OliverMadine** [comment, marked as answer]
Hi, 
Currently, the plugin is disabled by default since it does not always perform complete renames in multi-[component](https://docs.haskellstack.org/en/stable/build_command/#components) projects.
If you would like to enable it for use in a single-component project, then you would need to build HLS with the rename flag
(`cabal build --flag="+rename"`) and [configure your editor](https://haskell-language-server.readthedocs.io/en/latest/contributing/contributing.html#test-your-hacked-hls-in-your-editor) to use the resulting executable.

**Reply by m4dc4p** [reply]
Oh I see - I have a multi-component project so probably I should wait a bit longer. Thanks for replying!

**Reply by OliverMadine** [reply]
No problem. You can track the progress here: #2193.
